### PR TITLE
Code to set prefix mode based on account type

### PR DESCRIPTION
**Pull request by anmolanmol1234**
The code changes made are :-
If the account is FNS and has blob endpoint in uri, set the prefix mode to BLOB.
For other API's change the blob endpoint to dfs endpoint

### CODE REVIEW

**Review by saxenapranav — APPROVED** [review]
LGTM! Thanks for taking previous comments.
As discussed, we dont need to do getAcl and setAcl because they are required to fail in non-hns. At present as well it gives 400.

File: hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java [hunk 1]
```diff
@@ -190,9 +193,25 @@ public void initialize(URI uri, Configuration configuration)
...
+      /* since the filesystem has not been created. The API for HNS account would
+       * return 404 status.
+       */
+      if(ex.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+        isNamespaceEnabled = true;
+      } else {
+        throw ex;
+      }
+    }
+    if (!isNamespaceEnabled && uri.toString().contains(FileSystemUriSchemes.WASB_DNS_PREFIX)) {
```
Comment (sreeb-msft) [inline hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java:210]: Is the value of isNamespaceEnabled getting initialized anywhere before the getIsNamespaceEnabled call happens?

**Review by sreeb-msft — APPROVED** [review]

**Comment by sreeb-msft** [comment]
+1, approving!

File: hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java (hunk 1 above)
Comment (anmolanmol1234) [reply hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java:210]: This is the first call to determine, it's not getting called before this.

File: hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java [hunk 2]
```diff
@@ -928,7 +996,7 @@ public AbfsRestOperation checkAccess(String path, String rwx, TracingContext tra
     abfsUriQueryBuilder.addQuery(QUERY_PARAM_ACTION, CHECK_ACCESS);
     abfsUriQueryBuilder.addQuery(QUERY_FS_ACTION, rwx);
     appendSASTokenToQuery(path, SASTokenProvider.CHECK_ACCESS_OPERATION, abfsUriQueryBuilder);
-    URL url = createRequestUrl(path, abfsUriQueryBuilder.toString());
+    final URL url = createRequestUrl(path, abfsUriQueryBuilder.toString());
```
Comment (saxenapranav) [inline hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java:999]: lets check if if blob endpoint API and dfs endpoint API gives same exception -> statusCode, exceptionMessage.

Comment (anmolanmol1234) [reply hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java:999]: 2023-04-02 22:49:55,061 DEBUG [JUnit-testFsActionEXECUTE]: services.AbfsIoUtils (AbfsIoUtils.java:dumpHeadersToDebugLog(64)) -   HTTP Response=HTTP/1.1 403 This request is not authorized to perform this operation using this permission. 

The status code for blob endpoint is 403

Comment (anmolanmol1234) [reply hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java:999]: The response code and message is same for both blob and dfs endpoint.

**Review by saxenapranav — APPROVED** [review]
LGTM!

**Pull request merged**